src/main/java/com/emc/pravega/perf/PravegaPerfTest.java

```diff
@@ -259,131 +314,278 @@ private static void parseCmdLine(String[] args) {
+            this.messageSize = messageSize;      
             this.producer = factory.createEventWriter(streamName,
                     new JavaSerializer<String>(),
                     EventWriterConfig.builder().build());
 
-        }
+      }
 
-        /**
-         * This function will be executed in a loop and time behavior is measured.
-         * @return A function which takes String key and data and returns a future object.
-         */
-        BiFunction<String, String, CompletableFuture> sendFunction() {
+      /**
+       * This function will be executed in a loop and time behavior is measured.
+       * @return A function which takes String key and data and returns a future object.
+       */
+      BiFunction<String, String, CompletableFuture> sendFunction() {
             return  ( key, data) -> producer.writeEvent(key, data);
-        }
+      }
 
-        /**
-         * Executes the given method over the producer with configured settings.
-         * @param fn The function to execute.
-         */
-        void runLoop(BiFunction<String, String, CompletableFuture> fn) {
+      /**
+        * Executes the given method over the producer with configured settings.
+        * @param fn The function to execute.
+        */
+      void runLoop(BiFunction<String, String, CompletableFuture> fn) throws Exception {
 
             CompletableFuture retFuture = null;
-            for (int i = 0; i < secondsToRun; i++) {
-                int currentEventsPerSec = 0;
+            final long mSeconds = secondsToRun*1000;
+            long diffTime = mSeconds;
+
+            do {
 
                 long loopStartTime = System.currentTimeMillis();
-                while ( currentEventsPerSec < eventsPerSec) {
-                    currentEventsPerSec++;
+                for (int i = 0; i < eventsPerSec; i++)  {
 
                     // Construct event payload
                     String val = System.currentTimeMillis() + ", " + producerId + ", " + (int) (Math.random() * 200);
                     String payload = String.format("%-" + messageSize + "s", val);
+                    String key;
+                    if (isRandomKey) {
+                        key = Integer.toString(producerId + new Random().nextInt());
+                    } else {
+                        key = Integer.toString(producerId);
+                    }
+                   
                     // event ingestion
-                    long now = System.currentTimeMillis();
-                    retFuture = produceStats.runAndRecordTime(() -> {
-                                String key;
-                                if (isRandomKey) {
-                                    key = Integer.toString(producerId + new Random().nextInt());
-                                } else {
-                                    key = Integer.toString(producerId);
-                                }
+                    retFuture = stats.writeAndRecordTime(() -> {
```

Comment:
I think this code would be simpler with a Timer. See the class Timer in Pravega common.

Reply:
well for this i request i can take it in another PR please.
because , somewhere i read the System.nanoTime() is not rhread safe. need to check other alternative.
any change in this would affect the calculations done in the code.

Reply:
yes,not in this PR please, because lot of places there are System.currentTimeMillis is used and everything i have to change and latency , throughput calculation also chnages;  so , i request i will do this another PR